Tools/WebKitTestRunner/TestController.cpp

```diff
@@ -1464,17 +1464,26 @@ WKRetainPtr<WKStringRef> TestController::backgroundFetchState(WKStringRef)
 
 WKURLRef TestController::createTestURL(const char* pathOrURL)
 {
-    if (strstr(pathOrURL, "http://") || strstr(pathOrURL, "https://") || strstr(pathOrURL, "file://"))
+    if (strstr(pathOrURL, "http://") || strstr(pathOrURL, "https://"))
         return WKURLCreateWithUTF8CString(pathOrURL);
 
-    // Creating from filesytem path.
     size_t length = strlen(pathOrURL);
     if (!length)
         return 0;
 
+    if (length >= 7 && strstr(pathOrURL, "file://")) {
+        if (!WTF::FileSystemImpl::fileExists(String(pathOrURL + 7, length - 7))) {
```

Comment:
How much does this new `stat` slow down `run-webkit-tests`?

Reply:
Can we avoid this code path in "server" mode?